Bugfix - Uris are being deserialized like strings, and that's bad

shiftkey opened the pull request:
Let me know if there's anything missing from the PR

shiftkey committed:
added failing test for mapping to a Uri

shiftkey committed:
when a Uri is expected, validate it and populate it

shiftkey committed:
refactor tests for some readability

prabirshrestha commented on src/SimpleJson.Tests/DeserializeObjectTests.cs:
```diff
@@ -496,6 +496,26 @@ public void DeserializeDoubleQuotesCorrectly()
+            var json = "{\"url\":\"this is a broken uri\"}";
+            var result = SimpleJson.DeserializeObject<ClassWithUri>(json);
+
+            Assert.Null(result.url);
```
should be `IsNull`

haacked replied:
You're not in xUnit's world here Mr. @shiftkey. :stuck_out_tongue:

shiftkey replied:
![](http://i.imgur.com/N8X6XWq.gif)

shiftkey commented:
I've corrected that `Null` to `IsNull` to Ensure Nobody Freaks Out Any More About Unit Testing Frameworks.

I also added in a helper function which we've been using in Octokit.NET to keep things neat and tidy.

EDIT: you know what - we don't need that helper function.

prabirshrestha commented:
@shiftkey your changes are now in v0.30.0 (http://www.nuget.org/packages/SimpleJson/0.30.0)

Thanks for the PR
